### PR TITLE
Greasemonkey: add FAQ entry.

### PR DESCRIPTION
**Pull request by toofar**
The most common questsions regarding greasemonkey support on IRC are "how do I use it" and "why doesn't my script work", hopefully this can be a starting point for most people experiencing issues.

Hopefully it isn't to loooong.

I haven't used asciidoc before but I copied the other entries and it looks okay on my machine.

Happy now @jgkamat?

---
This change is [Reviewable](https://reviewable.io/reviews/qutebrowser/qutebrowser/3715)

### CODE REVIEW

File: doc/faq.asciidoc [hunk 1]
```diff
@@ -212,6 +212,35 @@ Why takes it longer to open an URL in qutebrowser than in chromium?::
...
+    section of the page displayed by `:version`. If you want to disable a
+    script just rename it, for example, to have `.disabled` on the end, after
+    the `.js` extension. To reload scripts from that directory run the command
+    `:greasemonkey-reload`.
++
+Troubleshooting: to check that your script is being loaded when
+`:greasemonkey-reload` runs you can start qutebrowser with the `--debug`
+argument and check the messages on the program's standard output with the
+`greasemonkey` facility. If there are javascript errors with your script you
+may also see messages with the `js` facility.
```
Comment (The-Compiler) [inline doc/faq.asciidoc]: Might want to suggest `--logfilter greasemonkey,js` here instead.

File: doc/faq.asciidoc [hunk 2]
```diff
@@ -212,6 +212,35 @@ Why takes it longer to open an URL in qutebrowser than in chromium?::
...
++
+Note that there are some missing features which you may run into:
+
+. Some scripts expect `GM_xmlhttpRequest` to ignore Cross Origin Resource
+  Sharing restrictions, this is currently not supported so scripts making
```
Comment (The-Compiler) [inline doc/faq.asciidoc]: I think there's a missing comma before "so" here?

File: doc/faq.asciidoc [hunk 3]
```diff
@@ -212,6 +212,35 @@ Why takes it longer to open an URL in qutebrowser than in chromium?::
...
+  requests to third party sites will often fail to function correctly.
+. If your backend is a QtWebEngine version 5.8, 5.9 or 5.10 regular expressions
```
Comment (The-Compiler) [inline doc/faq.asciidoc]: comma after 5.10

**Comment by The-Compiler** [comment]
Thanks!

File: doc/faq.asciidoc [hunk 4]
```diff
@@ -212,6 +212,37 @@ Why takes it longer to open an URL in qutebrowser than in chromium?::
...
+. Some scripts expect `GM_xmlhttpRequest` to ignore Cross Origin Resource
+  Sharing restrictions, this is currently not supported, so scripts making
+  requests to third party sites will often fail to function correctly.
+. If your backend is a QtWebEngine version 5.8, 5.9 or 5.10 then regular
+  expressions are not supported in `@include` or `@exclude` rules. If your
+  script uses them you can re-write them to use glob expressions or convert
+  them to `@match` rules.
+  See https://wiki.greasespot.net/Metadata_Block[the wiki] for more info.
+. Any greasemonkey API function to do with adding UI elements is not currently
+  supported. That means context menu extentensions and background pages.
```
Comment (toofar) [inline doc/faq.asciidoc]: >extentensions

Whoops, that has an extra ten in it.

**Comment by jgkamat** [comment]
yay! Hopefully this will make greasemonkey setup easier for everyone \o/

**Pull request merged**